assets/js/base/context/providers/cart-checkout/payment-methods/reducer.ts

```diff
@@ -28,17 +25,30 @@ const reducer = (
 		shouldSavePaymentMethod = false,
 		errorMessage = '',
 		paymentMethods = {},
+		paymentMethod = '',
 	}: ActionType
 ): PaymentMethodDataContextState => {
 	switch ( type ) {
+		case STATUS.PRISTINE:
+			return {
+				...DEFAULT_PAYMENT_DATA_CONTEXT_STATE,
+				currentStatus: STATUS.PRISTINE,
+				// This keeps payment method registration state and any set data. This effectively just resets the
+				// status and any error messages.
+				paymentMethods: {
+					...state.paymentMethods,
+				},
+				expressPaymentMethods: {
+					...state.expressPaymentMethods,
+				},
+				shouldSavePaymentMethod: state.shouldSavePaymentMethod,
+				activePaymentMethod: state.activePaymentMethod,
+				paymentMethodData: state.paymentMethodData,
+			};
```

Comment:
Instead of doing this, might as well
```suggestion
			return {
			// This keeps payment method registration state and any set data. This effectively just resets the
				// status and any error messages.
				...DEFAULT_PAYMENT_DATA_CONTEXT_STATE,
				...state,
				errorMessage: '',
				currentStatus: STATUS.PRISTINE,
			};
```

Reply:
This would make sure we don't forget to add a state here later.